File: 0796-rotate-string/0796-rotate-string.cpp
class Solution {
public:
    bool rotateString(string s, string goal) {
        s=s+s;
        int len=goal.length();

        bool value=false;

        for(int i=0;i<s.length();i++){
           string str=s.substr(i,len);
           if(str==goal && ((int)str.length()==((int)s.length()/2))) {     
                value=true;
                break; 
        }
        else 
        value=false;
        }
        return value;
    }
};

File: 0796-rotate-string/0796-rotate-string.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
         st=s+s
         stlen=len(st)
         value=False
         length=len(goal)

         for i in range(0,stlen):
             str=st[i:i+length]
             if str==goal and len(str)==(len(st))/2:
                 value=True
                 break
             else:
                 value=False    
    
         return value
          